internal/cmd/ktf/environments.go

```diff
@@ -210,14 +211,28 @@ func configureKongAddon(cmd *cobra.Command, envBuilder *environments.Builder) *e
 	if customGatewayImage != "" {
 		imageParts := strings.Split(customGatewayImage, ":")
 		if len(imageParts) == 1 {
-			imageParts[1] = "latest"
+			imageParts = append(imageParts, "latest")
 		}
 		if len(imageParts) != 2 { //nolint:gomnd
 			cobra.CheckErr(fmt.Errorf("malformed --kong-gateway-image: %s", customGatewayImage))
 		}
 		builder.WithProxyImage(imageParts[0], imageParts[1])
 	}
 
+	customControllerImage, err := cmd.PersistentFlags().GetString("kong-ingress-controller-image")
+	cobra.CheckErr(err)
+
+	if customControllerImage != "" {
+		imageParts := strings.Split(customControllerImage, ":")
+		if len(imageParts) == 1 {
+			imageParts = append(imageParts, "latest")
+		}
+		if len(imageParts) != 2 { //nolint:gomnd
+			cobra.CheckErr(fmt.Errorf("malformed --kong-ingress-controller-image: %s", customControllerImage))
+		}
+		builder.WithControllerImage(imageParts[0], imageParts[1])
```

Comment:
Well this doesn't look strictly right. If `len(imageParts) == 1` then we'll also get the error right? Maybe we should put this section in a `switch/case` statement?

Reply:
If len is 1 we append `latest` and len will be 2.

Reply:
Ah ok.

And then what happened is that it wasn't obvious to me that `cobra.CheckErr()` calls `os.Exit()`, hence I thought the program will continue with `len(imageParts) != 2`.

So this works but it's not easy to comprehend at first glance :) I'm ok with leaving this as is though 👍